clang/lib/Sema/SemaFunctionEffects.cpp

```diff
@@ -0,0 +1,1566 @@
+//=== SemaFunctionEffects.cpp - Sema handling of function effects ---------===//
```

Comment:
I've had a hard time with the review of this file, so I haven't done a close look.  Hopefully someone who understands this better than me has done a close review.